db/db_compaction_filter_test.cc

```diff
@@ -192,16 +197,28 @@ class KeepFilterFactory : public CompactionFilterFactory {
 
 class DeleteFilterFactory : public CompactionFilterFactory {
  public:
+  DeleteFilterFactory(TableFileCreationReason reason) : reason_(reason) {}
+
```

Comment:
I am having a little trouble following this class.  It looks like you can create it with a reason="Flush" but that option won't create a CompactionFilter?

Adding some comments as to why this is true to this class or what the intent would be helpful.

Reply:
> Adding some comments

Done.